Foundation/FileManager.swift

```diff
@@ -1441,73 +1441,63 @@ open class FileManager : NSObject {
         }
     }
 
-    @available(Windows, deprecated, message: "Not Yet Implemented")
     private func _removeItem(atPath path: String, isURL: Bool, alreadyConfirmed: Bool = false) throws {
         guard alreadyConfirmed || shouldRemoveItemAtPath(path, isURL: isURL) else {
             return
         }
 
-#if os(Windows)
-        NSUnimplemented()
-#else
-        if rmdir(path) == 0 {
-            return
-        } else if errno == ENOTEMPTY {
-
-            let stream = URL(fileURLWithPath: path).withUnsafeFileSystemRepresentation { (fsRep) -> UnsafeMutablePointer<FTS>? in
-                let ps = UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>.allocate(capacity: 2)
-                ps.initialize(to: UnsafeMutablePointer(mutating: fsRep))
-                ps.advanced(by: 1).initialize(to: nil)
-                defer {
-                    ps.deinitialize(count: 2)
-                    ps.deallocate()
+        let urlItem = URL(fileURLWithPath: path)
+        if urlItem.hasDirectoryPath {
+
+            let stream =  NSURLDirectoryEnumerator(
+              url: urlItem,
+              options: [],
+              errorHandler: {(url, err) in
+                self.shouldProceedAfterError(err,
+                                             removingItemAtPath: url.absoluteString,
+                                             isURL: true)
+              })
+            while let itemPath = stream.nextObject() as? String {
+              do {
+                guard alreadyConfirmed || shouldRemoveItemAtPath(itemPath, isURL: true) else {
+                  continue
                 }
-                return fts_open(ps, FTS_PHYSICAL | FTS_XDEV | FTS_NOCHDIR, nil)
-            }
-            
-            if stream != nil {
-                defer {
-                    fts_close(stream)
+                if (URL(fileURLWithPath: itemPath).hasDirectoryPath) {
+                  #if os(Windows)
+                  let result = _NS_rmdir(itemPath)
```

Comment:
Do we want to just fileprivate let _NS_rmdir = rmdir on non-Windows?

Reply:
And similarly for `_NS_unlink`.

Reply:
Or `fileprivate let rmdir = _NS_rundir` on Windows?

Reply:
Sure!